DataGrid: add Column Chooser and Header Filter demos (#2572)
DevExpress/devextreme-demos: JSDemos/Demos/DataGrid/ColumnChooser/description.md

If you want to allow users to change column visibility at runtime, set the **columnChooser**.[enabled](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/columnChooser/#enabled) property to **true**. To make sure that certain columns always stay visible, set the **columns[]**.[allowHiding](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/columns/#allowHiding) property to **false**. 

To display the column chooser, click the button in the toolbar above the DataGrid. You can specify a column chooser's position with the **columnChooser**.[position](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/columnChooser/#position) property. The way users show and hide columns depends on the **columnChooser**.[mode](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/columnChooser/#mode):

* *"dragAndDrop"*              
Users drag and drop column headers to and from the column chooser.

* *"select"*         

> Users select and deselect checkboxes with column names.

Reviewer: ```suggestion
Users select and deselect check boxes with column names.
```

> The column chooser selection is non-recursive. When a parent element is selected, nested elements remain unselected. If you want to select elements recursively, enable the **selection**.[recursive](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/columnChooser/select/#recursive) property.

Reviewer 2: First of all, I wanted a better connection to the previous paragraph - I wanted to mention that this is specific to only one of those modes. And then I'm just trying to simplify the text. 

```suggestion
In *"select"* mode, you can choose whether parent element selection affects child/nested elements. Use the **selection**.[recursive](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/columnChooser/select/#recursive) property.
```

> If the column chooser contains a lot of hidden column, you can enable column search UI. Assign **true** to the **search**.[enabled](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/columnChooser/search/#enabled) property. 

Reviewer: ```suggestion
If the column chooser contains many hidden columns, you can enable the column search UI. Assign **true** to the **search**.[enabled](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/columnChooser/search/#enabled) property.
```

> You can allow users to search through column names. For this, assign **true** to the **search**.[enabled](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/columnChooser/search/#enabled) property. 

Reviewer 2: "search through column names" - doesn't sound right. Maybe even sounds like I'm searching for parts of those names. 

```suggestion
If the column chooser contains a lot of hidden column, you can enable column search UI. Assign **true** to the **search**.[enabled](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/columnChooser/search/#enabled) property. 
```

> In this demo, use checkboxes below the DataGrid to toggle the search and selection features.

Reviewer: ```suggestion
In this demo, use the check boxes below the DataGrid to toggle the search and selection features.
```
